File: src/MixNetFx/FacFunctions/PrintClasses/FacPrintWorkGroupShiftDateBatch.vb
Imports Facesso.Data
Imports System.Drawing
Imports System.Collections.ObjectModel

Public Class FacPrintWorkGroupShiftDateBatch
    Inherits FacessoPrintBase

    Private myProductionPeriod As ProductionPeriod
    Private myWorkGroupList As Collection(Of WorkGroupInfo)

    Sub New(ByVal WorkGroupList As Collection(Of WorkGroupInfo), ByVal Period As ProductionPeriod, ByVal Username As String)
        MyBase.New("Schichtanalyse Produktiv-Site", "Detailaufstellung", Username)
        myProductionPeriod = Period
        myWorkGroupList = WorkGroupList
    End Sub

    Protected Overrides Sub PrepareDocument()
        MyBase.PrepareDocument(True)

        With PrintDocument
            'Mengentabelle der Produktiv-Site
            For Each locWorkGroup As WorkGroupInfo In myWorkGroupList
                For Each locItem As ProductionPeriodItem In myProductionPeriod
                    Dim locSdwr As New ShiftDateWorkResultInfo(New CombinedParametersInfo(locWorkGroup, locItem.ProductionDate, locItem.Shift))
                    PrintWorkGroupStatement(locSdwr)
                    .PageBreak()
                Next
            Next
        End With
    End Sub

    Protected Overridable Sub PrintWorkGroupStatement(ByVal sdwResult As ShiftDateWorkResultInfo)

        With PrintDocument
            .CurrentAlignment = ActiveDev.Printing.ADTextAlignment.Center
            .CurrentFont = LayoutAndNumberFormats.U2Font.ToFont
            .WriteLine("Schichtanalyse Produktiv-Site")
            .CurrentFont = LayoutAndNumberFormats.U2Font.ToFont
            .WriteLine(sdwResult.CombinedParameters.WorkGroup.ListItemText)
            .CurrentFont = LayoutAndNumberFormats.U3Font.ToFont
            .WriteLine("Schicht: " & sdwResult.CombinedParameters.Shift & "  -  " & sdwResult.CombinedParameters.ProductionDate.ToLongDateString).DistanceToNext = 20
            .WriteLine("Produktionsergebnis:").DistanceToNext = 10
            'Mengentabelle der Produktiv-Site
            .CurrentFont = LayoutAndNumberFormats.TableHeaderFont.ToFont
            .BeginTable(BorderStyle, 100, 300, 80, 100, 80, 80)
            .BuildTableHeader()
            .CurrentCellAlignment = ActiveDev.Printing.ADTextCellAlignment.TopCenter
            .WriteCells("AW-Nr.:", "REFA-Arbeitswert", "Menge", "Einheit", sdwResult.ProductionData.WorkGroup.BaseValueSynonym, "Summe")
            .BuildTableBody()
            .CurrentFont = LayoutAndNumberFormats.TextAndTableBodyFont.ToFont
            For Each locItem As ProductionDataItem In sdwResult.ProductionData
                .CurrentCellAlignment = ActiveDev.Printing.ADTextCellAlignment.TopRight
                .WriteCell(locItem.LabourValue.LabourValueNumber.ToString)
                .CurrentCellAlignment = ActiveDev.Printing.ADTextCellAlignment.TopLeft
                .WriteCell(locItem.LabourValue.LabourValueName)
                .CurrentCellAlignment = ActiveDev.Printing.ADTextCellAlignment.TopRight
                .WriteCell(locItem.Amount.ToString)
                .CurrentCellAlignment = ActiveDev.Printing.ADTextCellAlignment.TopLeft
                .WriteCell(locItem.LabourValue.Dimension.ToString)
                .CurrentCellAlignment = ActiveDev.Printing.ADTextCellAlignment.TopRight
                .WriteCell(locItem.LabourValue.TeHMin.ToString)
                .WriteCell(locItem.SubTotal.ToString)
            Next
            .EndTable()
            .WriteLine().DistanceToNext = 20
            .CurrentFont = LayoutAndNumberFormats.U3Font.ToFont
            .WriteLine("Beteiligte Mitarbeiter:").DistanceToNext = 10
            .CurrentFont = LayoutAndNumberFormats.TableHeaderFont.ToFont
            .BeginTable(BorderStyle, 80, 150, 85, 85, 70, 75, 75, 160)
            .BuildTableHeader()
            .CurrentCellAlignment = ActiveDev.Printing.ADTextCellAlignment.TopCenter
            .WriteCells("Pers.-Nr.:", "Name, Vorname", "Start", "Ende", "Pause", "Ausfall", "Handicap", "Zeitendelta")
            .BuildTableBody()
            .CurrentFont = LayoutAndNumberFormats.TextAndTableBodyFont.ToFont
            For Each locItem As EmployeeTimeLogInfoItem In sdwResult.EmployeeTimeLogItems
                .CurrentCellAlignment = ActiveDev.Printing.ADTextCellAlignment.TopRight
                .WriteCell(locItem.EmployeeInfo.PersonnelNumber.ToString)
                .CurrentCellAlignment = ActiveDev.Printing.ADTextCellAlignment.TopLeft
                .WriteCell(locItem.EmployeeInfo.LastName & ", " & locItem.EmployeeInfo.FirstName)
                .CurrentCellAlignment = ActiveDev.Printing.ADTextCellAlignment.TopCenter
                .WriteCell(locItem.ShiftStart.ToString("(ddd) HH:mm"))
                .WriteCell(locItem.ShiftEnd.ToString("(ddd) HH:mm"))
                .CurrentCellAlignment = ActiveDev.Printing.ADTextCellAlignment.TopRight
                .WriteCell(locItem.WorkBreak.ToString)
                .WriteCell(locItem.DownTime.ToString)
                .WriteCell(locItem.Handicap.ToString & " %")
                .CurrentCellAlignment = ActiveDev.Printing.ADTextCellAlignment.TopCenter
                .WriteCell(locItem.TimeDeltaStrings)
            Next
            .EndTable()
            .WriteLine()
            .CurrentFont = LayoutAndNumberFormats.U2Font.ToFont
            .WriteLine("Zusammenfassung").DistanceToNext = 10
            .CurrentFont = LayoutAndNumberFormats.TableHeaderFont.ToFont
            .BeginTable(ActiveDev.Printing.ADFrameCellBorderStyle.None, 200, 400)
            .CurrentCellAlignment = ActiveDev.Printing.ADTextCellAlignment.MiddleLeft
            .WriteCell("Produktiv-Site:") : .WriteCell(sdwResult.CombinedParameters.WorkGroup.ListItemText)
            .WriteCell("Datum:") : .WriteCell(sdwResult.CombinedParameters.ProductionDate.ToLongDateString)
            .WriteCell("Schicht:") : .WriteCell(sdwResult.CombinedParameters.Shift.ToString)
            .WriteCell("Minuten Referenz:") : .WriteCell(sdwResult.TotalReferenceIWT.ToString("#,##0.00"))
            .WriteCell("Minuten effektiv:") : .WriteCell(sdwResult.TotalEffectiveIWT.ToString("#,##0.00"))
            .WriteCell("Minuten effektiv (angepasst):") : .WriteCell(sdwResult.TotalEffectiveIWTAdj.ToString("#,##0.00"))
            .CurrentFont = LayoutAndNumberFormats.U3Font.ToFont
            .WriteCell(sdwResult.CombinedParameters.WorkGroup.IncentiveIndicatorSynonym)
            .WriteCell(sdwResult.DegreeOfTime.ToString(sdwResult.CombinedParameters.WorkGroup.IncentiveFormatString))
            .EndTable()
        End With
    End Sub
End Class

File: src/NetFxCSharp/Facesso.Functions/PrintClasses/FacPrintWorkGroupShiftDateBatch.cs
using Facesso.Data;
using Microsoft.VisualBasic;
using System;
using System.Collections;
using System.Collections.Generic;
using System.Collections.ObjectModel;
using System.Data;
using System.Diagnostics;
using System.Drawing;
using System.Linq;
using System.Threading.Tasks;
using System.Xml.Linq;

namespace Facesso.Functions
{
    public class FacPrintWorkGroupShiftDateBatch : FacessoPrintBase
    {
        private ProductionPeriod myProductionPeriod;
        private Collection<WorkGroupInfo> myWorkGroupList;
        public FacPrintWorkGroupShiftDateBatch(Collection<WorkGroupInfo> WorkGroupList, ProductionPeriod Period, string Username) : base("Schichtanalyse Produktiv-Site", "Detailaufstellung", Username)
        {
            myProductionPeriod = Period;
            myWorkGroupList = WorkGroupList;
        }

        protected override void PrepareDocument()
        {
            base.PrepareDocument(true);
            {
                var __with0 = PrintDocument;
                //Mengentabelle der Produktiv-Site
                foreach (WorkGroupInfo locWorkGroup in myWorkGroupList)
                {
                    foreach (ProductionPeriodItem locItem in myProductionPeriod)
                    {
                        ShiftDateWorkResultInfo locSdwr = new ShiftDateWorkResultInfo(new CombinedParametersInfo(locWorkGroup, locItem.ProductionDate, locItem.Shift));
                        PrintWorkGroupStatement(locSdwr);
                        __with0.PageBreak();
                    }
                }
            }
        }

        protected virtual void PrintWorkGroupStatement(ShiftDateWorkResultInfo sdwResult)
        {
            {
                var __with1 = PrintDocument;
                __with1.CurrentAlignment = ActiveDev.Printing.ADTextAlignment.Center;
                __with1.CurrentFont = LayoutAndNumberFormats.U2Font.ToFont();
                __with1.WriteLine("Schichtanalyse Produktiv-Site");
                __with1.CurrentFont = LayoutAndNumberFormats.U2Font.ToFont();
                __with1.WriteLine(sdwResult.CombinedParameters.WorkGroup.ListItemText);
                __with1.CurrentFont = LayoutAndNumberFormats.U3Font.ToFont();
                __with1.WriteLine("Schicht: " + sdwResult.CombinedParameters.Shift + "  -  " + sdwResult.CombinedParameters.ProductionDate.ToLongDateString()).DistanceToNext = 20;
                __with1.WriteLine("Produktionsergebnis:").DistanceToNext = 10;
                //Mengentabelle der Produktiv-Site
                __with1.CurrentFont = LayoutAndNumberFormats.TableHeaderFont.ToFont();
                __with1.BeginTable(BorderStyle, 100, 300, 80, 100, 80, 80);
                __with1.BuildTableHeader();
                __with1.CurrentCellAlignment = ActiveDev.Printing.ADTextCellAlignment.TopCenter;
                __with1.WriteCells("AW-Nr.:", "REFA-Arbeitswert", "Menge", "Einheit", sdwResult.ProductionData.WorkGroup.BaseValueSynonym, "Summe");
                __with1.BuildTableBody();
                __with1.CurrentFont = LayoutAndNumberFormats.TextAndTableBodyFont.ToFont();
                foreach (ProductionDataItem locItem in sdwResult.ProductionData)
                {
                    __with1.CurrentCellAlignment = ActiveDev.Printing.ADTextCellAlignment.TopRight;
                    __with1.WriteCell(locItem.LabourValue.LabourValueNumber.ToString());
                    __with1.CurrentCellAlignment = ActiveDev.Printing.ADTextCellAlignment.TopLeft;
                    __with1.WriteCell(locItem.LabourValue.LabourValueName);
                    __with1.CurrentCellAlignment = ActiveDev.Printing.ADTextCellAlignment.TopRight;
                    __with1.WriteCell(locItem.Amount.ToString());
                    __with1.CurrentCellAlignment = ActiveDev.Printing.ADTextCellAlignment.TopLeft;
                    __with1.WriteCell(locItem.LabourValue.Dimension.ToString());
                    __with1.CurrentCellAlignment = ActiveDev.Printing.ADTextCellAlignment.TopRight;
                    __with1.WriteCell(locItem.LabourValue.TeHMin.ToString());
                    __with1.WriteCell(locItem.SubTotal.ToString());
                }

                __with1.EndTable();
                __with1.WriteLine().DistanceToNext = 20;
                __with1.CurrentFont = LayoutAndNumberFormats.U3Font.ToFont();
                __with1.WriteLine("Beteiligte Mitarbeiter:").DistanceToNext = 10;
                __with1.CurrentFont = LayoutAndNumberFormats.TableHeaderFont.ToFont();
                __with1.BeginTable(BorderStyle, 80, 150, 85, 85, 70, 75, 75, 160);
                __with1.BuildTableHeader();
                __with1.CurrentCellAlignment = ActiveDev.Printing.ADTextCellAlignment.TopCenter;
                __with1.WriteCells("Pers.-Nr.:", "Name, Vorname", "Start", "Ende", "Pause", "Ausfall", "Handicap", "Zeitendelta");
                __with1.BuildTableBody();
                __with1.CurrentFont = LayoutAndNumberFormats.TextAndTableBodyFont.ToFont();
                foreach (EmployeeTimeLogInfoItem locItem in sdwResult.EmployeeTimeLogItems)
                {
                    __with1.CurrentCellAlignment = ActiveDev.Printing.ADTextCellAlignment.TopRight;
                    __with1.WriteCell(locItem.EmployeeInfo.PersonnelNumber.ToString());
                    __with1.CurrentCellAlignment = ActiveDev.Printing.ADTextCellAlignment.TopLeft;
                    __with1.WriteCell(locItem.EmployeeInfo.LastName + ", " + locItem.EmployeeInfo.FirstName);
                    __with1.CurrentCellAlignment = ActiveDev.Printing.ADTextCellAlignment.TopCenter;
                    __with1.WriteCell(locItem.ShiftStart.ToString("(ddd) HH:mm"));
                    __with1.WriteCell(locItem.ShiftEnd.ToString("(ddd) HH:mm"));
                    __with1.CurrentCellAlignment = ActiveDev.Printing.ADTextCellAlignment.TopRight;
                    __with1.WriteCell(locItem.WorkBreak.ToString());
                    __with1.WriteCell(locItem.DownTime.ToString());
                    __with1.WriteCell(locItem.Handicap.ToString() + " %");
                    __with1.CurrentCellAlignment = ActiveDev.Printing.ADTextCellAlignment.TopCenter;
                    __with1.WriteCell(locItem.TimeDeltaStrings);
                }

                __with1.EndTable();
                __with1.WriteLine();
                __with1.CurrentFont = LayoutAndNumberFormats.U2Font.ToFont();
                __with1.WriteLine("Zusammenfassung").DistanceToNext = 10;
                __with1.CurrentFont = LayoutAndNumberFormats.TableHeaderFont.ToFont();
                __with1.BeginTable(ActiveDev.Printing.ADFrameCellBorderStyle.None, 200, 400);
                __with1.CurrentCellAlignment = ActiveDev.Printing.ADTextCellAlignment.MiddleLeft;
                __with1.WriteCell("Produktiv-Site:");
                __with1.WriteCell(sdwResult.CombinedParameters.WorkGroup.ListItemText);
                __with1.WriteCell("Datum:");
                __with1.WriteCell(sdwResult.CombinedParameters.ProductionDate.ToLongDateString());
                __with1.WriteCell("Schicht:");
                __with1.WriteCell(sdwResult.CombinedParameters.Shift.ToString());
                __with1.WriteCell("Minuten Referenz:");
                __with1.WriteCell(sdwResult.TotalReferenceIWT.ToString("#,##0.00"));
                __with1.WriteCell("Minuten effektiv:");
                __with1.WriteCell(sdwResult.TotalEffectiveIWT.ToString("#,##0.00"));
                __with1.WriteCell("Minuten effektiv (angepasst):");
                __with1.WriteCell(sdwResult.TotalEffectiveIWTAdj.ToString("#,##0.00"));
                __with1.CurrentFont = LayoutAndNumberFormats.U3Font.ToFont();
                __with1.WriteCell(sdwResult.CombinedParameters.WorkGroup.IncentiveIndicatorSynonym);
                __with1.WriteCell(sdwResult.DegreeOfTime.ToString(sdwResult.CombinedParameters.WorkGroup.IncentiveFormatString));
                __with1.EndTable();
            }
        }
    }
}